modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp

```diff
@@ -21,6 +21,11 @@ namespace opencv_test
     {
         std::string method;
     };
+
+    struct CountStateSetupsParams
+    {
+        int* pSetupsCount = nullptr;
```

Comment:
weak_ptr?

Reply:
Thanks a lot for the catch!!
I 've chosen to not use exactly `weak_ptr`, because think that  `CountStateSetupsParams` is good  enough to be owner of passed memory
Fixed with `std::shared_ptr`, not `std::unique_ptr`, just to allow copying

Reply:
but if you indented to use copy then how's about mltithreading? do wee need to make operation ++ atomic?